source/state_representation/tests/testJointState.cpp

```diff
@@ -63,10 +68,98 @@ TEST(MultiplyByArray, PositiveNos) {
+  // same for the second initializer
+  StateRepresentation::JointState random2 =
+      StateRepresentation::JointState::Random("test",
+                                              std::vector<std::string>{"j0", "j1"});
+  // all data should be random (non 0)
+  EXPECT_TRUE(random2.get_positions().norm() > 0);
+  EXPECT_TRUE(random2.get_velocities().norm() > 0);
+  EXPECT_TRUE(random2.get_accelerations().norm() > 0);
+  EXPECT_TRUE(random2.get_torques().norm() > 0);
+}
+
+TEST(RandomPositionsInitialization, PositiveNos) {
+  StateRepresentation::JointPositions random = StateRepresentation::JointPositions::Random("test", 3);
+  // only position should be random
+  EXPECT_TRUE(random.get_positions().norm() > 0);
+  EXPECT_TRUE(random.get_velocities().norm() == 0);
+  EXPECT_TRUE(random.get_accelerations().norm() == 0);
+  EXPECT_TRUE(random.get_torques().norm() == 0);
+  // same for the second initializer
+  StateRepresentation::JointPositions random2 =
+      StateRepresentation::JointPositions::Random("test",
+                                              std::vector<std::string>{"j0", "j1"});
+  // all data should be random (non 0)
+  EXPECT_TRUE(random2.get_positions().norm() > 0);
+  EXPECT_TRUE(random2.get_velocities().norm() == 0);
+  EXPECT_TRUE(random2.get_accelerations().norm() == 0);
+  EXPECT_TRUE(random2.get_torques().norm() == 0);
+}
+
+TEST(RandomVelocitiesInitialization, PositiveNos) {
+  StateRepresentation::JointVelocities random = StateRepresentation::JointVelocities::Random("test", 3);
+  // only position should be random
+  EXPECT_TRUE(random.get_positions().norm() == 0);
+  EXPECT_TRUE(random.get_velocities().norm() > 0);
+  EXPECT_TRUE(random.get_accelerations().norm() == 0);
+  EXPECT_TRUE(random.get_torques().norm() == 0);
+  // same for the second initializer
+  StateRepresentation::JointVelocities random2 =
+      StateRepresentation::JointVelocities::Random("test",
+                                                  std::vector<std::string>{"j0", "j1"});
+  // all data should be random (non 0)
+  EXPECT_TRUE(random2.get_positions().norm() == 0);
+  EXPECT_TRUE(random2.get_velocities().norm() > 0);
+  EXPECT_TRUE(random2.get_accelerations().norm() == 0);
+  EXPECT_TRUE(random2.get_torques().norm() == 0);
+}
+
+TEST(RandomTorquesInitialization, PositiveNos) {
+  StateRepresentation::JointTorques random = StateRepresentation::JointTorques::Random("test", 3);
+  // only position should be random
+  EXPECT_TRUE(random.get_positions().norm() == 0);
+  EXPECT_TRUE(random.get_velocities().norm() == 0);
+  EXPECT_TRUE(random.get_accelerations().norm() == 0);
+  EXPECT_TRUE(random.get_torques().norm() > 0);
+  // same for the second initializer
+  StateRepresentation::JointTorques random2 =
+      StateRepresentation::JointTorques::Random("test",
+                                                   std::vector<std::string>{"j0", "j1"});
+  // all data should be random (non 0)
+  EXPECT_TRUE(random2.get_positions().norm() == 0);
```

Comment:
This doesn't correspond with the inline comment. Whats going on here? Same goes for the other test cases above btw.

Reply:
right bad copy pasting. Should be good now